File: CastIt.Android/lib/models/dtos/requests/play_file_request_dto.dart
import 'package:castit/models/dtos/base_socket_request_dto.dart';
import 'package:json_annotation/json_annotation.dart';

part 'play_file_request_dto.g.dart';

@JsonSerializable()
class PlayFileRequestDto extends AbstractBaseSocketRequestDto {
  @JsonKey(name: 'Id')
  final int id;

  @JsonKey(name: 'PlayListId')
  final int playListId;

  @JsonKey(name: 'Force')
  final bool force;

  PlayFileRequestDto({
    required this.id,
    required this.playListId,
    required this.force,
  }) : super();

  factory PlayFileRequestDto.fromJson(Map<String, dynamic> json) => _$PlayFileRequestDtoFromJson(json);

  Map<String, dynamic> toJson() => _$PlayFileRequestDtoToJson(this);
}

File: CastIt.Domain/Dtos/Requests/PlayFileRequestDto.cs
namespace CastIt.Domain.Dtos.Requests
{
    public class PlayFileRequestDto
    {
        public long Id { get; set; }
        public long PlayListId { get; set; }
        public bool Force { get; set; }
        public bool FileOptionsChanged { get; set; }
    }
}
